base/linalg/symmetric.jl

```diff
@@ -107,6 +107,21 @@ end
     r
 end
 
+function setindex!(A::Symmetric, v, i::Integer, j::Integer)
+    i == j || throw(ArgumentError("Cannot set a non-diagonal index in a symmetric matrix"))
```

Comment:
Yeah. We discussed that and it was considered too surprising. Are you considering a generic case where modifying the underlying buffer would be inconvenient?